File: create-dirs.py
import os
import sys

base_dir = r'C:\Users\Harshil\source\repos\harshil-sh\sql-performance-lab1\scenarios'

dirs = [
    os.path.join(base_dir, '04_window_functions'),
    os.path.join(base_dir, '05_keyset_pagination'),
    os.path.join(base_dir, '06_index_fragmentation')
]

for dir_path in dirs:
    os.makedirs(dir_path, exist_ok=True)
    gitkeep_path = os.path.join(dir_path, '.gitkeep')
    with open(gitkeep_path, 'w') as f:
        pass
    print(f'Created: {gitkeep_path}')

print('All directories created successfully')

File: create-dirs.js
const fs = require('fs');
const path = require('path');

const baseDir = 'C:\\Users\\Harshil\\source\\repos\\harshil-sh\\sql-performance-lab1\\scenarios';

const dirs = [
  path.join(baseDir, '04_window_functions'),
  path.join(baseDir, '05_keyset_pagination'),
  path.join(baseDir, '06_index_fragmentation')
];

dirs.forEach(dir => {
  fs.mkdirSync(dir, { recursive: true });
  fs.writeFileSync(path.join(dir, '.gitkeep'), '');
  console.log(`Created: ${dir}/.gitkeep`);
});

console.log('All directories created successfully');
